lib/main_common.pm

```diff
@@ -2616,6 +2616,9 @@ sub load_publiccloud_tests {
     elsif (get_var('PUBLIC_CLOUD_LTP')) {
         loadtest 'publiccloud/run_ltp';
     }
+    elsif (get_var('PUBLIC_CLOUD_SLES4SAP')) {
```

Comment:
@cfconrad it is getting bigger and bigger . I think we should consider refactor this to some generic way , WDYT ?

Reply:
From my point of view, the easiest way would be to specify which test-module will be loaded by a common variable. Similar to `EXTRATEST`. But we need to be aware, that currently most of these PC related test-modules are creating/deleting the instances within each test-module.
A other approach is what @pdostal is doing in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/8422

Reply:
hm , I briefly look into @pdostal PR now and I fail to catch the idea what is actually new in his approach ? As for me he just follow what we have currently.  I was thinking exactly about something like EXTRATEST approach. Also we may consider ideas from wicked/hpc approaches if they apply here , but this for sure not a part of **this** PR

Reply:
The new thing in his approach is, that he do not implement a test module like this:
```
1. create VM
2. do actual test
3. destroy vm
```
but he run multiple test-modules on one VM sequential in one JOB (this is like `EXTRATEST`)
``` 
[1st module] Create VM's and store them in some context (for cleanup and access)
[2nd module] Do some test
[3nd module] Do some more tests
[X module] Still doing more and more tests
[last module] destroy VM's
```

Reply:
So this is something that should be discussed apart of this PR, right? Is a poo# already opened (I didn't checked)?

Reply:
Yes we will do this in a different PR. Didn't found a poo# for this.

Reply:
Ok, I opened a [PR](https://progress.opensuse.org/issues/58310) for this. Feel free to add useful information or even change the title for something better :-)